core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c

```diff
@@ -0,0 +1,630 @@
+			     regu_lp_state[n].name);
+}
+
+/*
+ * Get PMIC and its I2C bus configuration from the device tree.
+ * Return 0 on success, 1 if no PMIC node found and a negative value otherwise
+ */
+static int dt_pmic_i2c_config(struct dt_node_info *i2c_info,
+			      struct stm32_pinctrl **pinctrl,
+			      size_t *pinctrl_count,
+			      struct stm32_i2c_init_s *init)
+{
+	int pmic_node = 0;
+	int i2c_node = 0;
+	void *fdt = NULL;
+	const fdt32_t *cuint = NULL;
+
+	fdt = get_embedded_dt();
+	if (!fdt)
+		return -FDT_ERR_NOTFOUND;
+
+	pmic_node = dt_get_pmic_node(fdt);
+	if (pmic_node < 0)
+		return 1;
+
+	cuint = fdt_getprop(fdt, pmic_node, "reg", NULL);
+	if (!cuint)
+		return -FDT_ERR_NOTFOUND;
+
+	pmic_i2c_addr = fdt32_to_cpu(*cuint) << 1;
+	if (pmic_i2c_addr > UINT16_MAX)
+		return -FDT_ERR_BADVALUE;
+
+	i2c_node = fdt_parent_offset(fdt, pmic_node);
+	if (i2c_node < 0)
+		return -FDT_ERR_NOTFOUND;
+
+	_fdt_fill_device_info(fdt, i2c_info, i2c_node);
+	if (!i2c_info->reg)
+		return -FDT_ERR_NOTFOUND;
+
+	return stm32_i2c_get_setup_from_fdt(fdt, i2c_node, init,
+					    pinctrl, pinctrl_count);
+}
+
+/*
+ * PMIC and resource initialization
+ */
+
+/* Return true if PMIC is available, false if not found, panics on errors */
+static bool initialize_pmic_i2c(void)
+{
+	int ret = 0;
+	struct dt_node_info i2c_info = { };
+	struct i2c_handle_s *i2c = &i2c_handle;
+	struct stm32_pinctrl *pinctrl = NULL;
+	size_t pin_count = 0;
+	struct stm32_i2c_init_s i2c_init = { };
+
+	ret = dt_pmic_i2c_config(&i2c_info, &pinctrl, &pin_count, &i2c_init);
+	if (ret < 0) {
+		EMSG("I2C configuration failed %d", ret);
+		panic();
+	}
+	if (ret)
+		return false;
+
+	/* Initialize PMIC I2C */
+	i2c->base.pa = i2c_info.reg;
+	i2c->base.va = (vaddr_t)phys_to_virt(i2c->base.pa, MEM_AREA_IO_SEC);
+	assert(i2c->base.va);
+	i2c->dt_status = i2c_info.status;
+	i2c->clock = i2c_info.clock;
+	i2c->i2c_state = I2C_STATE_RESET;
+	i2c_init.own_address1 = pmic_i2c_addr;
+	i2c_init.analog_filter = true;
+	i2c_init.digital_filter_coef = 0;
+
+	i2c->pinctrl = pinctrl;
+	i2c->pinctrl_count = pin_count;
+
+	stm32mp_get_pmic();
+
+	ret = stm32_i2c_init(i2c, &i2c_init);
+	if (ret) {
+		EMSG("I2C init 0x%" PRIxPA ": %d", i2c_info.reg, ret);
+		panic();
+	}
+
+	if (!stm32_i2c_is_device_ready(i2c, pmic_i2c_addr,
+				       PMIC_I2C_TRIALS,
+				       PMIC_I2C_TIMEOUT_BUSY_MS))
+		panic();
+
+	stpmic1_bind_i2c(i2c, pmic_i2c_addr);
+
+	stm32mp_put_pmic();
+
+	return true;
+}
+
+/*
+ * Automated suspend/resume at system suspend/resume is expected
+ * only when the PMIC is secure. If it is non secure, only atomic
+ * execution context acn get/put the PMIC resources.
+ */
+static TEE_Result pmic_pm(enum pm_op op, uint32_t pm_hint __unused,
+			  const struct pm_callback_handle *pm_handle __unused)
+{
+	if (op == PM_OP_SUSPEND)
+		stm32_i2c_suspend(&i2c_handle);
+	else
+		stm32_i2c_resume(&i2c_handle);
+
+	return TEE_SUCCESS;
+}
+
+KEEP_PAGER(pmic_pm);
```

Comment:
No blank line before `KEEP_PAGER(x);` please. I know that it will cause a `checkpatch` message, but all other instances have no blank line and @jenswi-linaro already mentioned a way to address this issue in https://github.com/OP-TEE/optee_os/pull/3377#issuecomment-564929705 (I've checked, it works, I can propose I PR).

Reply:
Ok. I'll fix.
Actually checkpatch complains when the label refers to a variable, not when it's a function.